backend/de/api/v1.py

```diff
@@ -219,26 +242,56 @@ def run_differential_expression(q: WmgQuery, criteria1, criteria2) -> Tuple[List
-    # this prevents overlapping populations from being compared
+    # identify number of overlapping populations
     filter_columns = [
         col
         for col in (base_expression_summary_indexed_dims + expression_summary_secondary_dims)
-        if col in es1.columns and col in es2.columns
+        if col in cell_counts1.columns and col in cell_counts2.columns
     ]
-    index1 = es1.set_index(filter_columns).index
-    index2 = es2.set_index(filter_columns).index
-    es2 = es2[~index2.isin(index1)]
-    if es2.shape[0] == 0:
-        return [], 1
+    index1 = cell_counts1.set_index(filter_columns).index
+    index2 = cell_counts2.set_index(filter_columns).index
+    overlap_filter = index1.isin(index2)
+    n_overlap = int(cell_counts1[overlap_filter]["n_total_cells"].sum())
+
+    """ (alec): If the specific overlapping populations are ever required, refer to this code
+    unique_overlap_indices = index1[overlap_filter].unique()
+    
+    num_values_per_level = {
+        name: unique_overlap_indices.get_level_values(name).unique().size for name in unique_overlap_indices.names
+    }
+    overlap = []
+    for dims in [dict(zip(unique_overlap_indices.names, idx, strict=False)) for idx in unique_overlap_indices]:
+        population = {}
+        if "disease_ontology_term_id" in dims and num_values_per_level["disease_ontology_term_id"] > 1:
+            population["disease_terms"] = ontology_term_id_label_mapping(dims["disease_ontology_term_id"])
+        if "sex_ontology_term_id" in dims and num_values_per_level["sex_ontology_term_id"] > 1:
+            population["sex_terms"] = ontology_term_id_label_mapping(dims["sex_ontology_term_id"])
+        if (
+            "self_reported_ethnicity_ontology_term_id" in dims
+            and "," not in dims["self_reported_ethnicity_ontology_term_id"]
+            and num_values_per_level["self_reported_ethnicity_ontology_term_id"] > 1
+        ):
+            population["self_reported_ethnicity_terms"] = ontology_term_id_label_mapping(
+                dims["self_reported_ethnicity_ontology_term_id"]
+            )
+        if "publication_citation" in dims and num_values_per_level["publication_citation"] > 1:
+            population["publication_citations"] = dims["publication_citation"]
+        if "cell_type_ontology_term_id" in dims and num_values_per_level["cell_type_ontology_term_id"] > 1:
+            population["cell_type_terms"] = ontology_term_id_label_mapping(dims["cell_type_ontology_term_id"])
+        if "tissue_ontology_term_id" in dims and num_values_per_level["tissue_ontology_term_id"] > 1:
+            population["tissue_terms"] = ontology_term_id_label_mapping(dims["tissue_ontology_term_id"])
+        if "organism_ontology_term_id" in dims and num_values_per_level["organism_ontology_term_id"] > 1:
+            population["organism_terms"] = ontology_term_id_label_mapping(dims["organism_ontology_term_id"])
+        overlap.append(population)
+    """
```

Comment:
I know this is just commented out code, but feels this can be significantly simplified for readability using a helper function like:

```
def add_population_term(population, term_id, label_mapping, term_name):
    if term_id in dims and num_values_per_level[term_id] > 1:
        population[term_name] = label_mapping(dims[term_id])
```


 and then something like:
 
 ```
     add_population_term(population, "disease_ontology_term_id", ontology_term_id_label_mapping, "disease_terms")
```
 and so on for the rest